src/validate.ts

```diff
@@ -36,18 +36,33 @@ function responseValidation(apiResponse: object, schema: object) {
 export function validate(req: object, apiResponse: object) {
   const responseType = req['inputs'][0]['intent'];
   if (responseType === 'action.devices.SYNC') {
+    //validate traits with attributes schema   
     return responseValidation(apiResponse, SYNC_RESPONSE_SCHEMA);
   } else if (responseType === 'action.devices.QUERY') {
+    //validate with states schema   
     return responseValidation(apiResponse, QUERY_RESPONSE_SCHEMA);
   } else if (responseType === 'action.devices.EXECUTE') {
-    const commands_arr = apiResponse['payload']['commands'];
-    const commandsLength= commands_arr.length;
-    //currently registering as an undefined api response. Needs fixes. 
-    for (let i = 0; i <= commandsLength; i++) {
-      const isOn = commands_arr[i]['states']['on'];
-      const isOnline = commands_arr[i]['states']['online'];
-      return [responseValidation(isOn, ON_OFF_STATES_SCHEMA), responseValidation(isOnline, ON_OFF_STATES_SCHEMA)];
+     //validate with states schema   
+    const commandsStatesExpect = {
+      'action.devices.commands.OnOff': ON_OFF_STATES_SCHEMA
     }
+    //gets the specific command  
+    const execution = req['inputs'][0]['payload']['execution'];
+    const executionLength = req['inputs'][0]['payload']['execution'].length;
+    
+    //identifies the part of the api response to validate against a schema 
+    const commands = apiResponse['payload']['commands'];
+    const commandsLength = commands.length;
+    
+    for (let i = 0; i >= executionLength; i++){
+        const commandName = execution[i]['command']
+        if (commandName in commandsStatesExpect){
+          for (let j = 0; j <= commandsLength; j++){
+              const states = commands[j]['states'];
+              return responseValidation(states, commandsStatesExpect[commandName]);
```

Comment:
If you return here, the function exits the first time and never goes lower.

Reply:
Yeah just saw that issue pop up, thinking about modifying the validate function to all take in multiple values and modify the response validation function to reflect those changes